### PR TITLE
Update server dependencies and add open api specification

### LINKED ISSUE #1382: Add Open API documentation (swagger)
The jhipster default template includes the Open API documentation as part of the server administration (using swagger). It would be good if we can (re)add this to Artemis (just in the same way as the default jhipster template does it). It would also be nice, if we could easily export the documentation and access it locally or on a website (i.e. not only during runtime).

### PR DESCRIPTION
**Pull request by krusche**
Medium updates in Spring Boot, Spring Cloud and Spring Messaging which led to some refactoring.
Also some minor client dependency updates
Fixes #1382

### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.

### Motivation and Context
We should stay up to date

### Description
Refactored some web socket related code, in particular moved logs and metrics into the MetricsBean to avoid dependency cycles within 

### Steps for Testing
1. Developers should test the development setup, starting the server and client application and communicating using web sockets on the localhost should still work as usual 
2. Testers should click through most views of the Artemis application and review if you notice issues

### Test Coverage
no changes

### Screenshots
no UI changes

### CODE REVIEW

**Review by FrankeLukas — COMMENTED** [review]
Still builds on my system and websockets work. I can still run the tests. So intermediate :+1: (since this is still labelled work in progress.)

One thing I noticed in the "Health" tab: There is a new `clientConfigServer` entry which has no translation and might not work properly (status is "unknown" and the details display an error).
grafik
grafik

**Comment by krusche** [comment]
yes, I need to translate this.
I am also still working on the Open API / Swagger integration

**Comment by krusche** [comment]
Fixed the translation issue

**Review by philgabr — COMMENTED** [review]
I had a look at all possible Artemis sites (exercise overview, exercise management, administration, exercise/exam creation & participation,...) and everything seems to work as intended :+1:

**Review by airelawaleria — APPROVED** [review]
Tested on ts1 with @simonzilker @marvinchimeka @JohannesWeiss. Works fine!

**Review by simonzilker — DISMISSED** [review]
Everything works as usual. Did not find any issue on ts1.

**Review by marvinchimeka — APPROVED** [review]
Could not find any issue. tested in testing session.

**Review by JohannesWeiss — APPROVED** [review]
Tested in testing session on ts1, did not spot any anomalies.

**Review by tiffca — APPROVED** [review]
Tested random scenarios and all looked good, no issues encountered!
created programming exercise, modeling, text, quiz, file upload 
tested participate, assess, complain 

user mgmt 
- add /remove group 

create/delete course
- register/deregister students 

exam 
- all exercise groups 
- participating 
- complaining 

Others
- checking scores
- lecture units 
- instructor course dashboard

**Review by valeriebucher — APPROVED** [review]
Looks good to me! Did not notice any issues.

**Review by EneaGore — DISMISSED** [review]
Looks good, couldnt ifnd any issues on ts1

File: src/main/webapp/app/shared/layouts/profiles/profile-info.model.ts [hunk 1]
```diff
@@ -6,6 +6,7 @@ export class ProfileInfo {
     activeProfiles: string[];
     ribbonEnv: string;
     inProduction: boolean;
+    public openApiEnabled?: boolean;
```
Comment (kloessst) [inline src/main/webapp/app/shared/layouts/profiles/profile-info.model.ts:9]:
```suggestion
    openApiEnabled?: boolean;
```

File: src/main/webapp/app/admin/admin.route.ts [hunk 2]
```diff
@@ -57,7 +58,14 @@ export const adminState: Routes = [
                 },
             },
             {
-                path: 'jhi-metrics',
+                path: 'docs',
```
Comment (kloessst) [inline src/main/webapp/app/admin/admin.route.ts:61]: I can still access this route directly even if the openapi profile is not active.

**Review by kloessst — COMMENTED** [review]
Code looks fine to me. See the other comments.

I participated in a programming exercise locally, and it worked fine.

File: src/main/webapp/app/admin/admin.route.ts (hunk 2 above)
Comment (krusche) [reply src/main/webapp/app/admin/admin.route.ts:61]: I guess this is ok for now. I’m not sure if we actually need this profile anyway

**Pull request merged**